src/python/grpcio_tests/tests/unit/_compression_test.py

```diff
@@ -52,75 +148,242 @@ def __init__(self, request_streaming, response_streaming):
+            client_function, {}, {}, {}, _GenericHandler(None), channel_kwargs,
+            multicallable_kwargs, server_kwargs, server_handler, _REQUEST)
+
+        if client_side_compressed:
+            self.assertCompressed(sent_ratio)
+        else:
+            self.assertNotCompressed(sent_ratio)
+
+        if server_side_compressed:
+            self.assertCompressed(received_ratio)
+        else:
+            self.assertNotCompressed(received_ratio)
+
+    def testDisableNextCompressionStreaming(self):
+        server_kwargs = {
+            'compression': grpc.Compression.Deflate,
+        }
+        _, received_ratio = _get_compression_ratios(
+            _stream_stream_client, {}, {}, {}, _GenericHandler(None), {}, {},
+            server_kwargs, _GenericHandler(disable_next_compression), _REQUEST)
+        self.assertNotCompressed(received_ratio)
+
+    def testDisableNextCompressionStreamingResets(self):
+        server_kwargs = {
+            'compression': grpc.Compression.Deflate,
+        }
+        _, received_ratio = _get_compression_ratios(
+            _stream_stream_client, {}, {}, {}, _GenericHandler(None), {}, {},
+            server_kwargs, _GenericHandler(disable_first_compression), _REQUEST)
+        self.assertCompressed(received_ratio)
+
+
+def _get_compression_str(name, value):
+    return '{}{}'.format(name, _COMPRESSION_NAMES[value])
+
+
+def _get_compression_test_name(client_streaming, server_streaming,
+                               channel_compression, multicallable_compression,
+                               server_compression, server_call_compression):
+    client_arity = 'Stream' if client_streaming else 'Unary'
+    server_arity = 'Stream' if server_streaming else 'Unary'
+    arity = '{}{}'.format(client_arity, server_arity)
+    channel_compression_str = _get_compression_str('Channel',
+                                                   channel_compression)
+    multicallable_compression_str = _get_compression_str(
+        'Multicallable', multicallable_compression)
+    server_compression_str = _get_compression_str('Server', server_compression)
+    server_call_compression_str = _get_compression_str('ServerCall',
+                                                       server_call_compression)
+    return 'test{}{}{}{}{}'.format(
+        arity, channel_compression_str, multicallable_compression_str,
+        server_compression_str, server_call_compression_str)
+
+
+def _test_options(remaining_options=_TEST_OPTIONS, options={}):
+    if len(remaining_options) == 0:
+        yield options
+    else:
+        key = next(iter(remaining_options.keys()))
+        new_remaining = dict(remaining_options)
+        del new_remaining[key]
+        values = remaining_options[key]
+        for value in values:
+            new_options = dict(options)
+            new_options[key] = value
+            for result in _test_options(new_remaining, new_options):
+                yield result
```

Comment:
My bad for suggesting generator. After searching, I found a Pythonic way to do this using [itertools.product](https://docs.python.org/2/library/itertools.html#itertools.product):
```Python
from itertools import product

def _test_options():
    test_option_names, test_option_choices = zip(*_TEST_OPTIONS.items())
    for test_option in product(*test_option_choices):
        yield zip(test_option_names, test_option)
```

Reply:
D'oh. `itertools.product` is what I should have reached for first instead of reimplementing it.